### PR TITLE
ToDoList:Crud operations

### PR DESCRIPTION
**Pull request by RileyManda**
I have completed the required implementation of the TodoList Applications Crud operations according to the stipulated [requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md):

The following features have been implemented:

**Implementations:**
- [x] All the elements of the user interface are be fully functional.
- [x] Removed all hardcoded items from the tasks array.
- [x] Implemented a function for adding a new task to the array.
- [x] Implemented a function for deleting a task from the array.
- [x] Implemented a function for editing task descriptions.
- [x] Ensured that by default new tasks have the property completed set to false.
- [x] Set the property index set to the value of the new array: When a new value is added index increments.[1..5..];
- [x] Implemented delete so that Deleting a task should update all remaining items and their indexes.
- [x] All the to-do items get saved to local storage.
- [x] Used ES6 modules to write modular JavaScript.

In addition, I have ensured the following functionality:

- [x] Project structure utilizes  Webpack that is based on the [Webpack exercise](https://github.com/RileyManda/webpack-demo) I completed.
- [x] Ensured that source files (index.html, index.js and style.css) and other files are located in /src directory.
- [x] Ensured that the distribution files are generated by webpack and served by webpack dev server from /dist folder.
- [x] When a user enters a list item and presses enter, a list item is added to the list
- [x]  Data is preserved in the browser's memory by using local storage.
- [x]  When an item is selected, the edit and delete icon is activated.
- [x] When Clear all completed is selected, all checked items are cleared from the list. 
- [x] Developed the UI to replicate the  [Example in the resuirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m1_list_structure.md)

In addition, this project also fulfills the  following general requirements:

* Used semantic HTML tags.
* Used CSS selectors correctly.
* Used JavaScript to manipulate DOM elements.
* Used JavaScript events.
* Used objects to store and access data.
* There are [no linter errors](https://github.com/microverseinc/linters-config).
* Used correct [GitHub flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md).
* Documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
* Followed the list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
* Followed the list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

Please kindly review my implementation and suggest any changes and improvements.

I look forward to your feedback.

Thank you
 Best regards,
 
 NB: There is currently an open issue related to a  bug in webhint and all webhint checks are failing.
 See the issue [here](https://github.com/webhintio/hint/issues/5614).

### CODE REVIEW

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @RileyManda,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ Linter checks passed successfully 👍🏽
✅ Add, edit, and remove features work fine 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [ ] Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique. This is not known at the moment because you're not saving todos indexes. Kindly work on that 👍🏽

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/index.js [hunk 1]
```diff
@@ -123,4 +145,7 @@ inputField.addEventListener('keydown', (event) => {
 });
 
 const clearButton = document.getElementById('clear');
-clearButton.addEventListener('click', clearCompletedItems);
\ No newline at end of file
+clearButton.addEventListener('click', () => {
+  clearCompletedItems();
+  saveListToStorage(getListFromDOM()); // Update storage after clearing completed items
+});
```
[the remark this answers is not in the document]
Comment (RileyManda) [reply src/index.js]: Hello @AmaduKamara 
Thank you for your review. I implemented the clear items feature based on the previous task which had a [video](https://www.youtube.com/watch?v=AcUd-_Yjjqg) showcasing the functionality of the application before crud operations and local storage. And in addition, this is my own code, it is not plagiarised.

File: src/addItem.js [hunk 2]
```diff
@@ -1,10 +1,9 @@
...
-} from './dragUtils.js';
+import { handleDragStart } from './dragUtils.js';
 import { handleCheckboxChange } from './clickUtils.js';
 import moreIcon from './assets/more-vert.png';
 import dustbinIcon from './assets/bin-icon.png';
 import './index.css';
+import { getListFromStorage, saveListToStorage } from './localStorage.js';
 
 const addItemToTodoList = () => {
   const inputField = document.querySelector('.add-item input');
```
[the remark this answers is not in the document]
Comment (RileyManda) [reply src/addItem.js:9]: Thank you for this feedback:
In this PR, I have already set the value of completed items to false.

Screen Shot 2023-06-14 at 3 25 56 PM

**Review by AdnanOlarmmi — APPROVED** [review]
Hi @RileyManda,

This is Adnan Is-haq, your beloved code reviewer, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Approved office man

## Highlights
- Fixed reviewer's suggestions.
- Nice app.

## Optional suggestions
_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**